tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,380 @@
+	provisioningConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, provisioningConfig)
+	nodeProviders := provisioningConfig.NodeProviders[0]
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+
+	testClusterConfig := clusters.ConvertConfigToClusterConfig(provisioningConfig)
+
+	if provisioningConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(client)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		provisioningConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if provisioningConfig.CNIs == nil {
+		provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+	nodeAndRoles := []provisioninginput.MachinePools{
+		provisioninginput.AllRolesMachinePool,
+	}
+	testClusterConfig.MachinePools = nodeAndRoles
+	testClusterConfig.KubernetesVersion = provisioningConfig.RKE2KubernetesVersions[0]
+	steveObject, err := provisioning.CreateProvisioningCustomCluster(client, &externalNodeProvider, testClusterConfig)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	provisioning.VerifyCluster(t, client, testClusterConfig, steveObject)
+
+	return steveObject, testClusterConfig, nil
+}
+
+func checkAWSS3Object(bucket string, key string) (bool, error) {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return false, err
+	}
+
+	s3Client := s3.NewFromConfig(sdkConfig)
+	_, err = s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func deleteAWSObject(bucket string, key string) error {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return err
+	}
+	svc := s3.NewFromConfig(sdkConfig)
+
+	input := &s3.DeleteObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+
+	_, err = svc.DeleteObject(context.TODO(), input)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func verifyRancherResources(client *rancher.Client, curUserList []*management.User, curProjList []*management.Project, curRoleList []*management.RoleTemplate) (error, error, error) {
+	var userErr error
+	var projErr error
+	var roleErr error
+
+	logrus.Info("Verifying user resources...")
+	for i := range curUserList {
+		user, err := users.GetUserIDByName(client, curUserList[i].Name)
+		if err != nil {
+			userErr = err
+		}
+		if user == "" {
+			userErr = fmt.Errorf("user not found")
+		}
+	}
+
+	logrus.Info("Verifying project resources...")
+	for i := range curProjList {
+		_, err := client.Management.Project.ByID(curProjList[i].ID)
+		if err != nil {
+			projErr = err
+		}
+	}
+
+	logrus.Info("Verifying role resources...")
+	for i := range curRoleList {
+		_, err := client.Management.RoleTemplate.ByID(curRoleList[i].ID)
+		if err != nil {
+			roleErr = err
+		}
+	}
+
+	return userErr, projErr, roleErr
+}
```

Comment:
Returning multiple errors isn't the most idiomatic - Would using errors.Join work here? Something like this:

```go
func verifyRancherResources(client *rancher.Client, curUserList []*management.User, curProjList []*management.Project, curRoleList []*management.RoleTemplate) error {
    var errs []error

    logrus.Info("Verifying user resources...")
    for _, user := range curUserList {
        userID, err := users.GetUserIDByName(client, user.Name)
        if err != nil {
            errs = append(errs, fmt.Errorf("user %s: %w", user.Name, err))
        } else if userID == "" {
            errs = append(errs, fmt.Errorf("user %s not found", user.Name))
        }
    }

    logrus.Info("Verifying project resources...")
    for _, proj := range curProjList {
        _, err := client.Management.Project.ByID(proj.ID)
        if err != nil {
            errs = append(errs, fmt.Errorf("project %s: %w", proj.ID, err))
        }
    }

    logrus.Info("Verifying role resources...")
    for _, role := range curRoleList {
        _, err := client.Management.RoleTemplate.ByID(role.ID)
        if err != nil {
            errs = append(errs, fmt.Errorf("role %s: %w", role.ID, err))
        }
    }

    return errors.Join(errs...)
}
```

Then something like this to use it:

```go
func (b *BackupTestSuite) TestS3InPlaceRestore() {
    
    // ********** OTHER CODE *********
    

    err := verifyRancherResources(b.client, userList, projList, roleList)
    require.NoError(b.T(), err)

    err = verifyRancherResources(b.client, userListPostBackup, projListPostBackup, roleListPostBackup)
    require.NoError(b.T(), err)
}

 // ********** OTHER CODE *********

```